packages/components/app/styles/components/side-nav/main.scss

```diff
@@ -66,56 +72,98 @@
 
 .hds-side-nav__menu-toggle-button {
   position: absolute;
+  top: -2px;
+  left: -1px;
   z-index: 1;
+  width: 24px;
+  height: 36px;
```

Comment:
Notice: the size of the border is influencing the position/alignment of the icon (probably it should be more resilient)

<img width="196" alt="screenshot_3081" src="https://github.com/hashicorp/design-system/assets/686239/39c78b79-7993-457b-9266-5925116389f4">